src/Forms/Prism.Forms/Navigation/PageNavigationService.cs

```diff
@@ -558,10 +558,9 @@ protected virtual async Task ProcessNavigationForNavigationPage(NavigationPage c
             }
 
             var clearNavigationStack = GetClearNavigationPageNavigationStack(currentPage);
-            var isEmptyOfNavigationStack = currentPage.Navigation.NavigationStack.Count == 0;
```

Comment:
This condition will always be false, because there is a check above and exiting the method.